File: test/fixtures/java-project/src/IAuthenticator.java
/**
 * Interface defining the contract for authentication services.
 */
public interface IAuthenticator {

    /**
     * Authenticate a user with the provided credentials.
     */
    public boolean authenticate(String username, String password);

    /**
     * Check whether a session is currently active.
     */
    public boolean isActive(String token);
}

File: test/fixtures/csharp-project/src/IAuthenticator.cs
namespace PureContext.Fixture
{
    /// <summary>
    /// Defines the contract for authentication services.
    /// </summary>
    public interface IAuthenticator
    {
        /// <summary>Authenticate with the provided credentials.</summary>
        bool Authenticate(string username, string password);

        /// <summary>Check whether the given session token is active.</summary>
        bool IsActive(string token);
    }
}
